blog/macos-rig/index.qmd

```diff
@@ -0,0 +1,309 @@
+* You’ve never heard of rig;
+* You want a one-stop-shop for setting up your rig next time;
+* You have a well-oiled, well-defined workflow but are interested in seeing how others set up their machines, to perhaps learn something new!
+
+Since it is the year of [ChatGPT](https://openai.com/blog/chatgpt/), [we did ask its suggested setup](#chat-gpt). We thought we could do better (at least in 2022, maybe not in 2024)! 🤗
+
+Let’s fall into a pit of success <sup>TM</sup> (when configuring macOS)!
+
+![](https://imgur.com/7J1bEaJ){fig-align="center" width="400" fig-alt="Hadley Wickham miming typing; in the background is a presentation that says My goal is to make a pit of success"}
+
+## Installation
+
+We’re going to use [Homebrew](https://brew.sh) to facilitate installation steps on macOS. Homebrew styles itself as 'the missing package manager for macOS.'
+
+1. Open Applications > Utilities > Terminal. Install Homebrew using the terminal command below, also provided on the Homebrew landing page. Paste this code into your terminal and hit enter:
+
+``` bash
+/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
+```
+
+2. Install [Git](https://git-scm.com). This version control system is going to come in handy for future projects on your new installation.
+
+``` bash
+brew install git
+```
+
+3. Configure [global Git settings](https://github.com/gvelasq/git-reference#setup).
+
+Run the below (replacing the text in the quotation marks with your information) to adjust your Git settings.
+
+```bash
+git config --global user.name "<full-name>"
+git config --global user.email "<email>"
+```
+
+4. Install [zsh](https://zsh.sourceforge.io).
+
+Z shell, or zsh, is a useful tool for interactive use in the Terminal. Several years back, macOS switched from using the bash shell to using zsh as the default shell. Installing the latest version of zsh from Homebrew gives you the *latest* version — otherwise, zsh and Git are both stuck at the version bundled with the last time you have updated macOS — which makes them open to vulnerabilities and/or don't have the latest features.
+
+```bash
+brew install zsh
+```
+
+5. Install [Oh My Zsh](https://ohmyz.sh/#install).
+
+Oh My Zsh (OMZ) is 'a delightful & open source framework for Zsh.' Or as ChatGPT succinctly put it:
+
+{{< tweet robbyrussell 1599596073282342914 >}}
+
+It comes with a ton of plugins, themes, and functions. For most users, having the great themes available from OMZ are enough to switch. But, the real value is [all the built-in OMZ plugins](https://github.com/ohmyzsh/ohmyzsh/wiki/Plugins) — which provide a rich set of aliases for Homebrew, Git, and many other command-line programs. For example, `gs` is `git status`, `bubo` is `brew update && brew outdated` (very helpful), etc. Here are some you should know:
```

Comment:
Suggest to change 'But,' to 'But for us,'.

Reply:
For all of the plugins mentioned below, instead of showing the full links to the plugins, suggest to hyperlink them to the word in the list. For example, hyperlink the brew plugin to the word 'brew' in the first bullet, the git plugin to the word 'git' in the second bullet, etc.